### ISSUE TITLE
CsvRoutines generating empty output with keepResourcesOpen

### ISSUE THREAD

**Issue by tanyona** [uniVocity/univocity-parsers#203]
With code similar to the following

```
CsvRoutines csvRoutines = new CsvRoutines(csvWriterSettings);
csvRoutines.setKeepResourcesOpen(true);

try (FileOutputStream fout = new FileOutputStream(fullOutputFileName)) {
	csvRoutines.write(rs, fout);
}
```

the output file generated is blank. If you remove the **setKeepResourcesOpen** line, the output is generated correctly in the file.

**Comment by jbax** [comment]
Thanks for reporting this one! I've fixed and released a 2.6.0-SNAPSHOT build with the adjustment.

Basically this happened because `.flush()` was not being called internally. When you use a `FileOutputStream` it is wrapped in a `Writer` object. That writer in turn was not closed or flushed when `keepResourcesOpen=true` and you have no access to it.

To make this work with the current version you just need to wrap your `FileOutputStream` in an `OutputStreamWriter`. You will then have control over the writer and can decide when to `flush()` or `close()` it.

**Comment by jbax** [comment]
Released version 2.5.9 with this bugfix (and others)

**Comment by tanyona** [comment]
OK. I've just tested this and it works fine. Thanks.
